### PR TITLE
Implement "Load More" buttons for domain registration step

### PR DESCRIPTION
**Pull request by jsnmoon**
This adds a link button labeled `Show more results` at the bottom of the suggestions list, like so: 

Show more results

Clicking this will append more results below the existing results (once this feature is supported in the API). Currently, no new suggestions are added.

This feature is implemented like so:

- When the user initially loads suggestions, we assume that page to be `page 1`.
- When the user clicks `Show more results`, we make a request for more suggestions with the `page_number` parameter set to `2`.
- Upon receiving the domain suggestion response, we append it to the existing list of search results and show it to the user.

# Testing Instructions
1. Spin up your environment locally, which has `domains/kracken-ui` feature flag enabled.
2. Navigate to [`/start/domains`](calypso.localhost:3000/start/domains).
3. Set your localStorage debug like so: `localStorage.debug = 'calypso:domains:register-domain-step'`.
4. Enter a query and ensure that you see a corresponding debug line in the console. (You can also double check with your network panel.)
5. Click `Show more results` and ensure that you see a corresponding debug line containing `page_number: 2` in the request parameter. (Likewise, this will also show up in the network panel.)

### CODE REVIEW

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -343,6 +347,16 @@ class RegisterDomainStep extends React.Component {
...
 				) }
 				{ this.content() }
+				{ isKrackenUI &&
+					! this.state.loadingResults && (
+						<Card
+							className="register-domain-step__next-page"
+							tagName="button"
+							onClick={ this.showNextPage }
+						>
+							Show more results
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: Needs to be wrapped in `translate`.

File: client/components/domains/register-domain-step/index.jsx [hunk 2]
```diff
@@ -343,6 +347,16 @@ class RegisterDomainStep extends React.Component {
 					/>
 				) }
 				{ this.content() }
+				{ isKrackenUI &&
+					! this.state.loadingResults && (
+						<Card
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: Inspired by https://github.com/Automattic/wp-calypso/pull/23368, I've checked how navigating this new element by `Tab` looks like:


Looks like it's missing focus state, even though the backing element for the link is a `button`? Probably best/easiest to try to replicate what `Use a domain I own` link uses? (unless it's some sort of hack ;))

File: client/components/domains/register-domain-step/index.jsx (hunk 1 above)
Comment (jsnmoon) [reply client/components/domains/register-domain-step/index.jsx]: Oops, right on.

File: client/components/domains/register-domain-step/index.jsx (hunk 2 above)
Comment (jsnmoon) [reply client/components/domains/register-domain-step/index.jsx]: Definitely something wrong - I'll investigate this.

**Comment by jsnmoon** [comment]
Thanks for the review @klimeryk! I'll work on the issues you mentioned and follow up here.

To summarize, I'll work on the following:
- Don't show the button in the Domain Management page.
- Don't show the button upon first page load in the signup flow.
- Only show the loader when applicable:
	- Don't show loader over subdomain suggestion
 	- Don't show loader over existing suggestions
- Don't hide "Show more results" button when loading more suggestions
- Wrap "Show more results" in translate.
- Fix a11y focus highlight

**Comment by klimeryk** [comment]
All sound good except:
> Don't show the button in the Domain Management page.

Show it, but not when the default suggestions are shown (unless it _is_ possible to fetch more - since they are fetched from Domainsbot as well, AFAIK)

**Comment by jsnmoon** [comment]
Addressed review feedback and added a little more polish to the overall UI. Ready for another look.

File: client/components/domains/register-domain-step/index.jsx [hunk 3]
```diff
@@ -343,12 +350,40 @@ class RegisterDomainStep extends React.Component {
...
 				{ this.content() }
+				{ this.renderPaginationControls() }
 				{ queryObject && <QueryDomainsSuggestions { ...queryObject } /> }
 				<QueryContactDetailsCache />
 			</div>
 		);
 	}
 
+	renderPaginationControls() {
+		const isKrackenUI = config.isEnabled( 'domains/kracken-ui' );
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: `isKrackenUi` 🙏 See https://github.com/Automattic/wp-calypso/pull/23384#discussion_r178966246

File: client/components/domains/register-domain-step/index.jsx [hunk 4]
```diff
@@ -646,24 +704,24 @@ class RegisterDomainStep extends React.Component {
...
+		debug( 'onSearch handler was triggered with query', searchQuery );
 		const domain = getFixedDomainSearch( searchQuery );
 
 		this.setState(
 			{ lastQuery: searchQuery, lastSurveyVertical: this.props.surveyVertical },
 			this.save
 		);
 
-		if ( ! domain || ! this.state.loadingResults ) {
+		if ( ! domain || ( typeof domain === 'string' && domain.trim() === '' ) ) {
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: When is this case possible? Seems `getFixedDomainSearch` above takes care of trimming and such (and in general ensuring it's a proper string).

File: client/components/domains/register-domain-step/index.jsx (hunk 3 above)
Comment (jsnmoon) [reply client/components/domains/register-domain-step/index.jsx]: D'oh.

File: client/components/domains/register-domain-step/index.jsx (hunk 4 above)
Comment (jsnmoon) [reply client/components/domains/register-domain-step/index.jsx]: Having looked at `getFixedDomainSearch`, I don't think it's possible. I'll simplify this to `domain === ''`.

**Comment by jsnmoon** [comment]
> Something that needs to be addressed, though, is that when requesting a new "page" of results in NUX, we're also re-requesting the WPCOM subdomain suggestion (and actually updating it). 

Fixed! If the changes look sane to you, please mark this PR ready to merge again :)

**Comment by klimeryk** [comment]
Thanks - tested and the new changes look good 👍 
...but while testing I noticed a regression creeped in at some point: if you change any of the filters, the page number does not reset to `1` anymore :(

**Comment by jsnmoon** [comment]
Thanks Igor! 🎉

**Pull request merged**